src/TMap.cpp

```diff
@@ -2173,7 +2173,11 @@ void TMap::downloadMap(const QString& remoteUrl, const QString& localFileName)
     }
 
     if (localFileName.isEmpty()) {
-        mLocalMapFileName = mudlet::getMudletPath(mudlet::profileXmlMapPathFileName, mProfileName);
+        if (url.toString().endsWith(QLatin1String("xml"))) {
+            mLocalMapFileName = mudlet::getMudletPath(mudlet::profileXmlMapPathFileName, mProfileName);
+        } else {
+            mLocalMapFileName = mudlet::getMudletPath(mudlet::profileMapPathFileName, mProfileName, QStringLiteral("map.dat"));
```

Comment:
Will a downloaded *binary* map file **always** be called `map.dat` - I guess this requires it?

Reply:
Incidentally do the previous error messages need to be revised to handle the case when it is not an XML file that is being downloaded now?

Reply:
Yes, binary files will be `map.dat`, just like xml ones are `map.xml`.